src/main/java/com/google/maps/internal/PolylineEncoding.java

```diff
@@ -35,9 +35,7 @@
 
     int len = encodedPath.length();
 
-    // For speed we preallocate to an upper bound on the final length, then
```

Comment:
It was half valid :) ArrayList.trimToSize() is a Java7 feature, so that part was false. The rest seems obvious enough from the code though, so I dropped it.